### PR TITLE
Interview notes

### LINKED ISSUE #24: Saving notes
### Description
- As a user, I'd like to add and save my interview notes

### Acceptance Criteria
- Add a button to allow user to save notes
- On click of button save notes to db
- On load of dashboard, display saved notes in textarea. if any.

### PR DESCRIPTION
**Pull request by beebzz**
Fixes #24

### CODE REVIEW

File: graphql/api.js [hunk 1]
```diff
@@ -256,6 +257,38 @@ export const updateApplication = async (id, company, role, currentExcitement) =>
...
+};
+
+export const updateInterviewNotes = async (id, notes, date, type) => {
+  const query = `mutation updateInterviewNotes($id: ID!, $notes: String!, $date: String!, $type: String!) {
+    updateInterview(id: $id, data:{
+      date: $date
+      type: $type
+      notes: $notes
+      }){
+    			notes
```
Comment (zeeberry) [inline graphql/api.js:273]: ah i see what you mean about the indentation. I wonder if its a tabs vs spaces issue.

File: graphql/api.js [hunk 2]
```diff
@@ -256,6 +257,38 @@ export const updateApplication = async (id, company, role, currentExcitement) =>
...
+    errorMessage: getErrorMessage(null, data)
+  };
+};
+
+export const updateInterviewNotes = async (id, notes, date, type) => {
+  const query = `mutation updateInterviewNotes($id: ID!, $notes: String!, $date: String!, $type: String!) {
+    updateInterview(id: $id, data:{
+      date: $date
+      type: $type
+      notes: $notes
```
Comment (zeeberry) [inline graphql/api.js:271]: does it need `date` and `type` to update?

File: pages/dashboard.js [hunk 3]
```diff
@@ -2,25 +2,43 @@ import { useState, useEffect } from 'react';
...
+  const onChange = (e) => {
+    setNotes(e.target.value);
+  };
+
+  const saveInterviewNotes = async (id, date, type) => {
+    const { data, errorMessage } = await updateInterviewNotes(id, notes, date, type);
+    if (data?.updateInterview.notes === notes) {
+      toast.success("Notes saved! 🚀");
+    }
+    if(!!errorMessage) {
```
Comment (zeeberry) [inline pages/dashboard.js:37]: there's nothing wrong with your syntax here, however it is not very common in JS. JS uses type coercion when a value is in a boolean context like this one. Basically it automatically transforms the value to a boolean, so we dont really have a need to prefix a value with `!!`. In this case you can just do `if(errorMessage)`. More info [here](https://developer.mozilla.org/en-US/docs/Glossary/Truthy).

File: pages/dashboard.js [hunk 4]
```diff
@@ -71,6 +89,8 @@ export default function Dashboard() {
                           <div>Date: {interview.date}</div>
                           <div>Nerves: {interview.nerves}</div>
                           <div>Excitement: {interview.excitement}</div>
+                          <textarea defaultValue={interview.notes} onChange={onChange} />
```
Comment (zeeberry) [inline pages/dashboard.js:92]: what happens when there are newlines added in the textarea? do they persist after they get saved?

**Review by zeeberry — COMMENTED** [review]
good work! I added some comments and questions.

File: graphql/api.js (hunk 2 above)
Comment (beebzz) [reply graphql/api.js:271]: Yeah, for some reason GraphQL forces you to include a type's required props (in this case date and type) even if you're updating only one

File: pages/dashboard.js (hunk 3 above)
Comment (beebzz) [reply pages/dashboard.js:37]: Yeah I'm aware haha, I just find `!!` easier to interpret most of the time (though admittedly i'm not too consistent with it). I don't mind changing this though if you prefer

File: pages/dashboard.js (hunk 4 above)
Comment (beebzz) [reply pages/dashboard.js:92]: Yeah they're saved as newline characters

File: graphql/api.js (hunk 1 above)
Comment (beebzz) [reply graphql/api.js:273]: same, the linter will hopefully fix this

File: graphql/api.js (hunk 2 above)
Comment (zeeberry) [reply graphql/api.js:271]: ok. interesting.

File: pages/dashboard.js (hunk 4 above)
Comment (zeeberry) [reply pages/dashboard.js:92]: dope!

**Review by zeeberry — APPROVED** [review]
LGTM!

**Pull request merged**